### DISCUSSION TITLE
ItemGroup categories seem to not exist

### DISCUSSION THREAD

**Discussion by hyphanova** [FabricMC/fabric-api#2739] [Mod Dev Support] [answered]
I've been trying to work on a random mod, and I've gotten to the point of attempting to make an item.

Following a tutorial that seems to match up with the fabric wiki, I've tried adding in the line of code to make a new item and put it under an item category, in this case I am trying to put it in `COMBAT` using the `new FabricItemSettings()` method.

This code for this file is shown below, but for whatever reason the part the defines the item category, in this case the word `COMBAT`, is giving an error, saying "COMBAT cannot be resolved or is not a field _Java(33554502)_."
```java
package v0ru5.tactical.tadpoles;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class TacticalTadpoles implements ModInitializer {
	// This logger is used to write text to the console and the log file.
	// It is considered best practice to use your mod id as the logger's name.
	// That way, it's clear which mod wrote info, warnings, and errors.
	public static final Logger LOGGER = LoggerFactory.getLogger("modid");

	public static final Item TADPOLE_LAUNCHER = new Item(new FabricItemSettings().group(ItemGroup.COMBAT));

	@Override
	public void onInitialize() {
		// This code runs as soon as Minecraft is in a mod-load-ready state.
		// However, some things (like resources) may still be uninitialized.
		// Proceed with mild caution.

		LOGGER.info("Hello Fabric world!");
	}
}
```

I don't know what is causing this error, and why it just isn't working at all, as the game fails to load up with this error. I'm using Visual Studio Code for this, and it also doesn't show any autocomplete categories after `ItemGroup` as it usually would with anything else.

That's about it, any help would be much appreciated.

**Comment by Juuxel** [comment, marked as answer]
The wiki tutorial is written for 1.19.2. 1.19.3 overhauled how item groups work, so you no longer specify it in the item settings. The vanilla item group instances also moved to a new class, `ItemGroups`.

To add items to a vanilla item group in 1.19.3, you can use `ItemGroupEvents.modifyEntriesEvent`. For example:
```java
ItemGroupEvents.modifyEntriesEvent(ItemGroups.COMBAT).register(entries -> {
    entries.add(TADPOLE_LAUNCHER);
});
```

**Reply by hyphanova** [reply]
Thank you for the code help, I wasn't aware of the item changes with 1.19.3.
I've spent a little bit of time attempting to implement it into my code, and I'm unsure of where to put it. I don't know if I'm missing something, but I've tried replacing the old line of code with this newer one and I'm unable to get it working properly.
I'm still new to working with Fabric, and I'll keep trying to work on this, but I appreciate you responding to this

**Reply by haykam821** [reply]
You should register the event handler during mod initialization, so within the `onInitialize` method will work.

**Reply by hyphanova** [reply]
That works, thank you.
One final thing, I can't figure out how to create the item itself, as adding items to categories works but `TADPOLE_LAUNCHER` is not recognized as a variable, so I'm assuming I have to separately define it as an item.

**Reply by mangomeep** [reply]
Something like this should work, `public static final Item TADPOLE_LAUNCHER = Registry.register(Registries.ITEM, new Identifier(TacticalTadpoles.MOD_ID, "tadpole_launcher"), new Item(new FabricItemSettings()));` 

From my understanding, you'd also need to have the corresponding model, textures, and translations.
